### DISCUSSION TITLE
Creation of additional end point based on another column

### DISCUSSION THREAD

**Discussion by baskarkanini** [Azure/data-api-builder#2458] [Q&A] [answered]
HI Team,
This is my table and used the DAB CLI commands for this.
dab add Pets --source "[home].[Pets]"--source.type "table" --source.key-fields "PetId" --permissions "anonymous:*"

CREATE TABLE [home].[Pets](
	[PetId] [int] IDENTITY(1,1) NOT NULL,
	[PersonId] [int] NOT NULL,
	[IsOwner] [bit] NOT NULL,
	[HouseId] [int] NULL,
	[PetName] [nvarchar](100) NULL,
	[PetType] [nvarchar](100) NULL,
	[PetBreed] [nvarchar](100) NULL,
	[CreatedDate] [datetime] NULL,
	[CreatedBy] [nvarchar](100) NULL,
	[ModifiedDate] [datetime] NULL,
	[ModifiedBy] [nvarchar](100) NULL,
	[IsActive] [bit] NULL)

This is working fine. I dont have any issue. 
I got get, getbyid, put, post, patch and delete.
https://localhost:5001/api/Pets/PetId/3

The problem:
GetByPetID. Similar to this how can i create another endpoint for get by housed.
eventhough this is not a primary key.

Please help me with the DAB command.

My alternate option would be stored procedure. But i prefer the CLI command which will solve my problem.

Thank you for your time and responses.

**Comment by sezal98** [comment, marked as answer]
You can use Stored Procedures for this task. Alternatively, you can use filter option in the REST Get request for getting the same result
Attaching docs for reference
Filter in REST : https://learn.microsoft.com/en-us/azure/data-api-builder/rest#filter
Stored Procedure: https://learn.microsoft.com/en-us/azure/data-api-builder/database-objects#stored-procedures

**Reply by myassociationmanager** [reply]
Thank you @sezal98 .
I have used the filter for this one and it working fine. 
Also I have created a stored procedure and it is working fine.
